tools/state-viewer/src/congestion_control.rs

```diff
@@ -59,4 +68,72 @@ impl CongestionControlCmd {
             println!("{:?} - {:?}", shard_id, congestion_info);
         }
     }
+
+    fn run_benchmark(&self, home_dir: &Path, near_config: &NearConfig, store: Store) {
+        let (epoch_manager, runtime, state_roots, block_header) =
+            load_trie(store, home_dir, near_config);
+
+        let prev_hash = block_header.prev_hash();
+
+        let epoch_id = epoch_manager.get_epoch_id_from_prev_block(prev_hash).unwrap();
+        let protocol_config = runtime.get_protocol_config(&epoch_id).unwrap();
+        let runtime_config = protocol_config.runtime_config;
+
+        for (shard_id, &state_root) in state_roots.iter().enumerate() {
+            let shard_id = shard_id as u64;
+            let shard_uid = epoch_manager.shard_id_to_uid(shard_id, &epoch_id).unwrap();
+
+            let tries = runtime.get_tries();
+
+            let state_root = Self::add_receipts(tries, shard_uid, state_root);
+            let trie = runtime.get_trie_for_shard(shard_id, prev_hash, state_root, true).unwrap();
+
+            let now = std::time::Instant::now();
+            let congestion_info =
+                bootstrap_congestion_info(&trie, &runtime_config, shard_id).unwrap();
+            let duration = now.elapsed();
+
+            println!("{:?} - {:?}", shard_id, congestion_info);
+            println!("Boostrapping took {:?}", duration);
+        }
+    }
+
+    fn add_receipts(tries: ShardTries, shard_uid: ShardUId, state_root: StateRoot) -> StateRoot {
+        let trie = tries.get_trie_for_shard(shard_uid, state_root);
+        let mut trie_update = TrieUpdate::new(trie);
+        let mut queue = DelayedReceiptQueue::load(&trie_update).unwrap();
+
+        for _ in 0..10_000 {
+            let receipt = Self::create_receipt();
+            queue.push(&mut trie_update, &receipt).unwrap();
+        }
+
+        trie_update.commit(StateChangeCause::UpdatedDelayedReceipts);
+        let (_, trie_changes, _) = trie_update.finalize().unwrap();
+
+        let mut store_update = tries.store_update();
+        tries.apply_all(&trie_changes, shard_uid, &mut store_update);
+
+        trie_changes.new_root
+    }
+
+    fn create_receipt() -> Receipt {
+        let predecessor_id = "predecessor".parse().unwrap();
+        let receiver_id = "receiver".parse().unwrap();
+
+        // TOOD - generate random data
+        let data = vec![0, 1, 2, 3];
```

Comment:
This breaks for some reason if I change it to 
```
  let data = vec![0; 900];
```
but seems to work fine for sizes lower than 900 bytes